scripts/install_prereqs.py

```diff
@@ -0,0 +1,453 @@
+
+    Returns:
+        (str): Absolute path to vcpkg executable.
+    """
+    if is_windows_os():
+        vcpkg_executable_file_path = os.path.join(get_vcpkg_root_dir(), 'vcpkg.exe')
+    elif is_linux_os() or is_mac_os():
+        vcpkg_executable_file_path = os.path.join(get_vcpkg_root_dir(), 'vcpkg')
+    else:
+        raise ValueError('Unsupported operating system (supported os: windows, mac, linux')
+
+    return vcpkg_executable_file_path
+
+
+def get_vcpkg_response_file_path(vcpkg_triplet):
+    """ Get absolute path to vcpkg response file.
+    Response file stores the list of packages to install with vcpkg.
+
+    Args:
+        vcpkg_triplet (str): Name of the vcpkg triplet (eg: 'x64-windows-static', 'x64-osx').
+
+    Returns:
+        (str): Absoulte path to vcpkg response file.
+    """
+    vcpkg_response_file_path = os.path.join(ROOT_REPO_DIR, 'external', 
+                                            'vcpkg_' + vcpkg_triplet + '_response_file.txt')
+    return vcpkg_response_file_path
+
+
+def get_vcpkg_installation_script_path():
+    """ Get absolute path to the script used to build and install vcpkg.
+
+    Raises:
+        ValueError: Unsupported operating system.
+
+    Returns:
+        (str): Absolute path to vcpkg installation script.
+    """
+    if is_windows_os():
+        script_absolute_path = os.path.join(get_vcpkg_root_dir(), 'bootstrap-vcpkg.bat')
+    elif is_linux_os() or is_mac_os():
+        script_absolute_path = os.path.join(get_vcpkg_root_dir(), 'bootstrap-vcpkg.sh')
+    else:
+        raise ValueError('Unsupported operating system (supported os: windows, mac, linux')
+
+    return script_absolute_path
+
+
+def is_tool_installed(tool):
+    """Check if a tool is installed on the system.
+
+    Args:
+        tool (str): Name of the tool.
+
+    Returns:
+        (bool) : True if installed on the system, False otherwise.
+    """
+    return distutils.spawn.find_executable(tool)
+
+
+def is_windows_os():
+    """Check if the current OS is Windows.
+
+    Returns:
+        (bool): True if windows, False otherwise.
+    """
+    return sys.platform.startswith('win')
+
+
+def is_linux_os():
+    """Check if the current OS is Linux.
+
+    Returns:
+        (bool): True if windows, False otherwise.
+    """
+    return sys.platform.startswith('linux')
+
+
+def is_mac_os():
+    """Check if the current OS is mac.
+
+    Returns:
+        (bool): True if windows, False otherwise.
+    """
+    return sys.platform.startswith('darwin')
+
+
+def get_system_package_manager():
+    """ Get package manger based on the platform.
+    brew for mac, apt for linux, choco for windows.
+
+    Raises:
+        ValueError: If the current OS could not be identified.
+
+    Returns:
+        (str): Package manager for current platform.
+    """
+    if is_windows_os():
+        if not is_tool_installed(PackageManager.WINDOWS):
+            raise ValueError('Did not find {0} on the system. \
```

Comment:
This will capture the big block of whitespace in front of "Please install" and "try setting" and produce something like this:

```
"Did not find hi on the system.                                 Please install it or                                 try setting install_if_not_found=True"
```

Python will concatenate consecutive strings automatically, so you can write something like this:

```
raise ValueError('a'
                 ' b'
                 ' c'.format(stuff))
```

I've seen this kind of thing in a couple of places, and concatenation (implicit, as above, or explicit, with +) is strongly recommended over using \